src/kixi/heimdall/handler.clj

```diff
@@ -30,96 +24,29 @@
-            (catch clojure.lang.ExceptionInfo e
-              (do (log/debug "Unsign refresh token failed")
-                  nil)))))
-
-(defn make-refresh-token [issued-at-time auth-conf user]
-  (let [exp (-> (t/plus (t/now) (t/days 30)) (c/to-long))]
-    (jwt/sign {:user-id (:id user)}
-              (pkey auth-conf)
-              {:alg :rs256 :iat issued-at-time :exp exp})))
-
-(defn make-token-pair! [session auth-conf user]
-  (let [issued-at-time (c/to-long (t/now))
-        refresh-token (make-refresh-token issued-at-time auth-conf user)]
-    (refresh-token/add! session {:refresh-token refresh-token
-                                 :issued issued-at-time
-                                 :user-id (:id user)})
-    {:token-pair {:auth-token (make-auth-token user auth-conf)
-                  :refresh-token refresh-token}}))
-
-(defn create-auth-token [session auth-conf credentials]
-  (let [[ok? res] (user/auth session credentials)]
-    (if ok?
-      [true (make-token-pair! session auth-conf (:user res))]
-      [false res])))
-
 (defn auth-token [req]
-  (let [[ok? res] (create-auth-token (:cassandra-session (:components req))
-                                     (:auth-conf req)
-                                     (:params req))]
+  (let [[ok? res] (service/create-auth-token (:cassandra-session (:components req))
+                                             (:auth-conf req)
+                                             (:params req))]
     (if ok?
       {:status 201 :body res}
       {:status 401 :body res})))
 
 
-(defn refresh-auth-token [session auth-conf refresh-token]
-  (if-let [unsigned (unsign-token auth-conf refresh-token)]
-    (let [user-uuid (java.util.UUID/fromString (:user-id unsigned))
-          refresh-token-data (refresh-token/find-by-user-and-issued session
-                                                                    user-uuid
-                                                                    (:iat unsigned))
-          user (user/find-by-id session user-uuid)]
-      (if (:valid refresh-token-data)
-        (do
-          (refresh-token/invalidate! session (:id refresh-token-data))
-          [true (make-token-pair! session auth-conf user)])
-        [false {:message "Refresh token revoked/deleted or new refresh token already created"}]))
-    [false {:message "Invalid or expired refresh token provided"}]))
-
-
-
-(defn refresh-auth-token-route [req]
+(defn refresh-auth-token [req]
   (let [refresh-token (-> req :params :refresh-token)
-        [ok? res] (refresh-auth-token (:cassandra-session (:components req))
-                                      (:auth-conf req)
-                                      refresh-token)]
+        [ok? res] (service/refresh-auth-token (:cassandra-session (:components req))
+                                              (:auth-conf req)
+                                              refresh-token)]
     (if ok?
       {:status 201 :body res}
```

Comment:
Still not sure if refreshing should cause a 201 - does it actually return a new auth token?